src/Nethermind/Nethermind.Blockchain/Receipts/IReceiptFinder.cs

```diff
@@ -9,7 +9,7 @@ namespace Nethermind.Blockchain.Receipts
     public interface IReceiptFinder
     {
         Hash256? FindBlockHash(Hash256 txHash);
-        TxReceipt[] Get(Block block, bool recover = true);
+        TxReceipt[] Get(Block block, bool recover = true, bool recoverSender = true);
```

Comment:
Maybe would be better to collapse `recover` and `recoverSender` into an enum?

```
public enum ReceiptRecoveryOptions { None, TransactionAndBlock, TransactionAndBlockWithSender }
```

Hard to name

Reply:
I'd say, don't recover sender at all in receipt finder. Its a weird behaviour.